RecoTracker/TkTrackingRegions/plugins/PixelInactiveAreaFinder.cc

```diff
@@ -0,0 +1,1104 @@
+                       innerSpan.zSpan.first, innerSpan.zSpan.second);
+    areas.emplace_back(outerSpan.rSpan.first, outerSpan.rSpan.second,
+                       outerSpan.phiSpan.first, outerSpan.phiSpan.second,
+                       outerSpan.zSpan.first, outerSpan.zSpan.second);
+    ret.emplace_back(std::move(areas), std::move(item.second));
+  }
+
+  return ret;
+}
+
+std::vector<std::pair<std::vector<PixelInactiveAreaFinder::DetGroupSpan>, std::vector<PixelInactiveAreaFinder::LayerSetIndex> > >
+PixelInactiveAreaFinder::InactiveAreas::spansAndLayerSets(const GlobalPoint& point, float zwidth) const {
+  // TODO: in the future use 2D-r for the origin for the phi overlap check
+  const float zmin = point.z()-zwidth;
+  const float zmax = point.z()+zwidth;
+
+  std::vector<std::pair<std::vector<DetGroupSpan>, std::vector<LayerSetIndex> > > ret;
+
+  LogDebug("PixelInactiveAreaFinder") << "Origin at " << point.x() << "," << point.y() << "," << point.z() << " z half width " << zwidth;
+
+  for(LayerSetIndex i=0, end=inactiveLayerPairIndices_->size(); i<end; ++i) {
+    const auto& layerIdxPair = (*inactiveLayerPairIndices_)[i];
+    const auto& innerSpans = inactiveSpans_[layerIdxPair.first];
+    const auto& outerSpans = inactiveSpans_[layerIdxPair.second];
+
+    for(const auto& innerSpan: innerSpans) {
+      for(const auto& outerSpan: outerSpans) {
+
+        if(phiRangesOverlap(innerSpan.phiSpan, outerSpan.phiSpan)) {
+          std::pair<float,float> range(0,0);
+
+          bool zOverlap = false;
+          const auto innerDet = std::get<0>((*inactiveLayers_)[layerIdxPair.first]);
+          const auto outerDet = std::get<0>((*inactiveLayers_)[layerIdxPair.second]);
+          if(innerDet == GeomDetEnumerators::PixelBarrel) {
+            if(outerDet == GeomDetEnumerators::PixelBarrel)
+              zOverlap = getZAxisOverlapRangeBarrel(innerSpan, outerSpan, range);
+            else
+              zOverlap = getZAxisOverlapRangeBarrelEndcap(innerSpan, outerSpan, range);
+          }
+          else {
+            if(outerDet == GeomDetEnumerators::PixelEndcap)
+              zOverlap = getZAxisOverlapRangeEndcap(innerSpan, outerSpan, range);
+            else
+              throw cms::Exception("LogicError") << "Forward->barrel transition is not supported";
+          }
+
+          if(zOverlap && zmin <= range.second && range.first <= zmax) {
+#ifdef EDM_ML_DEBUG
+            Stream ss;
+            for(auto ind: (*layerSetIndexInactiveToActive_)[i]) {
+              ss << ind << ",";
+            }
+            ss << "\n  ";
+            detGroupSpanInfo(innerSpan, ss);
+            ss << "\n  ";
+            detGroupSpanInfo(outerSpan, ss);
+            LogTrace("PixelInactiveAreaFinder") << " adding areas for active layer sets " << ss.str();
+#endif
+            ret.emplace_back(std::vector<DetGroupSpan>{{innerSpan, outerSpan}}, (*layerSetIndexInactiveToActive_)[i]);
```

Comment:
here and for areasAndLayerSets the first argument is size 2. Can it be made to an array [or another fixed size selection] instead of a vector?
... ignore this if there is another modification that can extend this past 2

Reply:
I had this in mind as well but apparently forgot. I don't think there is any apparent need for anything beyond 2 (as that would mean we would try to mitigate cases with >= 3 inactive layers; ok possible for phase2 FPix but I would ignore that for now). Size 1 could be useful for "triplet mitigation for HLT" planned by @JanFSchulte, so I'll make it `edm::VecArray` to have consistent interface.